Factory.Cargo/CargoFactoryApi.cs

```diff
@@ -0,0 +1,36 @@
+using Factory.Cargo.Factory;
+using Factory.Cargo.TransportationFactory;
+
+namespace Factory.Cargo;
+
+public class CargoFactoryApi
+{
+    private Dictionary<string, ITransportationFactory> _transportationFactories = new();
+
+    public void AddBaseFactories()
+    {
+        _transportationFactories.Add("Air",new AirTransportationFactory());
+        _transportationFactories.Add("Train",new TrainTransportationFactory());
+        _transportationFactories.Add("Ship",new ShipTransportationFactory());
+    }
+    
+    
+    public void AddTransportationFactory(string mode, ITransportationFactory transportationFactory)
+        => _transportationFactories.Add(mode, transportationFactory);
+
+    public ITransportation Create(string mode, string? origin = null, string? destination = null)
```

Comment:
if you have 10 factories and each of them needs different parameter. so how to solve it ? are you going to add 10 different null parameter?

Reply:
@mohammadKarimi 
I've developed new way to pass parameters dynamically.
`
public class CargoFactoryService
{
 public ITransportation Create(Dictionary<string,object> parameters);
}
`

so the use now can call factory in this way
` public LibraryUser()
    {
        _factoryService.AddTransportationFactory("Truck", new TruckTransportationFactory());

        var truckTransportation = _factoryService.Create("Truck", new Dictionary<string, object>
        {
            { "TruckType", "Trailer" },
        });
    }`